apis/externalsecrets/v1alpha1/externalsecret_types.go

```diff
@@ -130,6 +130,54 @@ type ExternalSecretDataRemoteRef struct {
 	Property string `json:"property,omitempty"`
 }
 
+// ExternalSecretDataFromRemoteRef defines Provider data location.
+type ExternalSecretDataFromRemoteRef struct {
+	// Used to select a specific version and property from the secret
+	// +optional
+	Extract ExternalSecretExtract `json:"extract,omitempty"`
+	// Used to find secrets based on tags or regular expressions
+	// +optional
+	Find ExternalSecretFind `json:"find,omitempty"`
+}
+
+func (ref ExternalSecretDataFromRemoteRef) GetDataRemoteRef() ExternalSecretDataRemoteRef {
+	return ExternalSecretDataRemoteRef{
+		Key:      ref.Extract.Key,
+		Property: ref.Extract.Property,
+		Version:  ref.Extract.Version,
+	}
+}
+
+type ExternalSecretExtract struct {
+	// Key is the key used in the Provider
+	// +optional
+	Key string `json:"key,omitempty"`
+
+	// Used to select a specific version of the Provider value, if supported
+	// +optional
+	Version string `json:"version,omitempty"`
+
+	// +optional
+	// Used to select a specific property of the Provider value (if a map), if supported
+	Property string `json:"property,omitempty"`
+}
```

Comment:
```suggestion
func (ref ExternalSecretDataFromRemoteRef) GetDataRemoteRef() ExternalSecretDataRemoteRef {
	return ExternalSecretDataRemoteRef{
		Key:      ref.Extract.Key,
		Property: "",
		Version:  ref.Extract.Version,
	}
}

type ExternalSecretExtract struct {
	// Key is the key used in the Provider
	// +optional
	Key string `json:"key,omitempty"`

	// Used to select a specific version of the Provider value, if supported
	// +optional
	Version string `json:"version,omitempty"`
}
```
The idea here is that we are not supposed to use `property` in `dataFrom`, so now that we are separating the structs, we might as well give it explicitly.

Reply:
Actually AzureKeyVault does use the Property in the dataFrom... nevermind.

Reply:
@gusfcarvalho  The `Property` on `GetSecretMap` is to get the specified property from a key? Isn't every provider using it like that?

Reply:
Property from a key only happens at Data, not dataFrom. In DataFrom I think it gets ignored on all providers. We probably can remove this field from dataFrom then.

Yeah, azure was the only provider that had something similar to getallsecrets and sparker the conversation, and we should keep getallsecrets logic only on getallsecrets.

Reply:
Wait, what if I have this json as a secret?
```json
{
  "Name": "External",
  "LastName": "Secret",
  "Address": {
    "Street": "Myroad st.",
    "CP": "J4K4T4"
  }
}
```

I'm using this ExternalSecret to get the Address

```yaml

  dataFrom:
  - extract:
      key: json-secret
      property: Address
```

And this is returning a map with like the following:
```go
smtc.expectedData["Street"] = []byte("Myroad st.")
smtc.expectedData["CP"] = []byte("J4K4T4")
```

That is actually working and it's being [tested](https://github.com/external-secrets/external-secrets/blob/main/pkg/provider/azure/keyvault/keyvault_test.go#L300) with the Azure Key Vault provider.

Isn't this supposed to happen?

Reply:
Did you have a look over the other providers? I think azure is the only one doing that in dataFrom

Reply:
From the other providers perspectives, AWS and GCP would be compliant with this use of the property. Only vault would not be compliant, but that could be easily changed by changing this method here:  https://github.com/external-secrets/external-secrets/pull/613/files#diff-503805349eecb8259761e5c39bbcf2987ec284b9522173c4d230d8bf3b9ce6e5R169

I really like the outcome of that. I think we can leave this as is and change vault's behavior so the experience is the closest as possible with the other providers =)

Reply:
@moolen @gusfcarvalho @sebagomez I vote for removing the `property` field from the dataFrom use cases altogether. It is too confusing to leave it there. We only have examples documented for AKV and the full snippet. And I feel like this is not something that we want to maintain (?)

Reply:
@knelasevero you mentioned in our slack that somebody could implement this behavior with a template. Can you share how? I think having an alternative for this example is a good enough reason to remove the property

Reply:
I'd vote to keep, improve docs with some examples, and improve HashiVault to have the same feature.

Reasons: It is a feature that is already implemented. There is no extra code needed to maintain. The reason this happens everywhere is essentially because in most part of the providers the GetSecretMap uses GetSecret(ref) to then map the JSON if the data is parseable. Sure, It is a weird behavior and could be done with `data` getting the Key + templating, but it is easier to do with dataFrom.

Reply:
Ok, I just asked some of our users that I know that use a bunch of our features, and they already use property for dataFrom  in GCP! haha

So if users are using this even without the docs mentioning it nowhere, I guess they really want this.

Yeah, then I stay corrected haha.